Invalid C code for inline iterator with inner proc

While trying out the different nim solutions to the Hamming numbers task from RosettaCode I found that one variant did not compile. Reducing it to the max gave me this:

### 
```nim
iterator log_nodups_hamming(): int {.inline.} =
  let lb3 = 1
  proc mul3(): int = lb3
  yield mul3()

for h in log_nodups_hamming():
  break
for h in log_nodups_hamming():
  break
```

### Output
```
/mnt/space/data/nim/.cache/hn_5_test/d/@mhn_5_test.nim.c: In function ‘NimMainModule’:
/mnt/space/data/nim/.cache/hn_5_test/d/@mhn_5_test.nim.c:214:5: error: ‘colonenv_’ undeclared (first use in this function); did you mean ‘colonenv__2’?
  214 |   (*colonenv_).lb31 = ((NI) 1);
      |     ^~~~~~~~~
      |     colonenv__2
/mnt/space/data/nim/.cache/hn_5_test/d/@mhn_5_test.nim.c:214:5: note: each undeclared identifier is reported only once for each function it appears in
```
This is the expansion of the second loop.

### Additional Information

* Original source: https://rosettacode.org/wiki/Hamming_numbers#Imperative_iterator_implementation_of_the_above_functional_version
* Using `--gc:arc` the compilation succeeds.

#### Used versions
```
$ nim -v
Nim Compiler Version 1.2.0 [Linux: amd64]
Compiled at 2020-04-03
Copyright (c) 2006-2020 by Andreas Rumpf

active boot switches: -d:release
```

* The 1.2.1 nightly from 28.4. and the 1.3.1 nightly from 29.4. show the same problem.

ok, closing the other one in favor of this one; there is some complementary information so I'm pasting the issue here:

-----------------------
reduced from https://github.com/nim-lang/Nim/pull/18535#issuecomment-882957657 /cc @brightly-salty 

### Example
```nim
# iterator envPairs*(): int {.closure.} = # Error: internal error: environment misses: foo
iterator envPairs*(): int =
  var foo: seq[int]
  proc fun() =
    foo = @[]
  fun()
  yield 3

proc main() =
  for a in envPairs():
    discard
  for a in envPairs():
    discard
static: main() # nim c in vm: ok
# static: main() # nim js in vm: Error: cannot evaluate at compile time: foo
main() # nim c: @mt12609.nim.c:372:5: error: use of undeclared identifier 'colonenv_'; did you mean 'colonenv__2'?
# main() # nim js: # Error: internal error: symbol has no generated name: foo
```

### Current Output
```
/@mt12609.nim.c:372:5: error: use of undeclared identifier 'colonenv_'; did you mean 'colonenv__2'?
                (*colonenv_).foo1 = (tySequence__qwqHTkRvwhrRyENtudHQ7g*)0;
                  ^~~~~~~~~
                  colonenv__2
/Users/timothee/git_clone/nim/timn/build/nimcache/@mt12609.nim.c:364:65: note: 'colonenv__2' declared here
                tyObject_Env_t12609dotnim_envPairs___Dwg4Sd9awieN34NfR6DSgaw* colonenv__2;
                                                                              ^
/Users/timothee/git_clone/nim/timn/build/nimcache/@mt12609.nim.c:375:46: error: use of undeclared identifier 'colonenv_'; did you mean 'colonenv__2'?
                T4_.ClP_0 = fun_t4950544857_3; T4_.ClE_0 = colonenv_;

```
### Expected Output
works

### Additional Information
1.5.1 73ce40aaf71196faa08072216bd826a94b5d40b1

--- a/doc/contributing.rst
+++ b/doc/contributing.rst
@@ -10,6 +10,9 @@ Contributing
 
 Contributing happens via "Pull requests" (PR) on github. Every PR needs to be
 reviewed before it can be merged and the Continuous Integration should be green.
+The title of a PR should contain a brief description. If it fixes an issue,
+in addition to the number of the issue, the title should also contain a description 
+of the issue.
 
 The PR has to be approved by two core developers or by Araq.
 

